src/store/auth/auth-reducer.ts

```diff
@@ -4,42 +4,42 @@
 
 import { authActions, AuthAction } from "./auth-action";
 import { User } from "../../models/user";
-import { authService } from "../../services/services";
+import { ServiceRepository } from "../../services/services";
 import { removeServerApiAuthorizationHeader, setServerApiAuthorizationHeader } from "../../common/api/server-api";
 
 export interface AuthState {
     user?: User;
     apiToken?: string;
 }
 
-export const authReducer = (state: AuthState = {}, action: AuthAction) => {
+export const authReducer = (services: ServiceRepository) => (state: AuthState = {}, action: AuthAction) => {
     return authActions.match(action, {
         SAVE_API_TOKEN: (token: string) => {
-            authService.saveApiToken(token);
+            services.authService.saveApiToken(token);
```

Comment:
I think that we should handle side effects such as calling service methods in actions or middlewares as we do in other features. Reducer should not cause any side effects.  https://redux.js.org/basics/reducers#handling-actions

Reply:
ok, will try to refactor that as well!